engine/hash_table.cpp

```diff
@@ -47,8 +47,8 @@ bool HashTable::MatchHashEntry(const Slice &key, uint32_t hash_k_prefix,
 Status HashTable::Search(const KeyHashHint &hint, const Slice &key,
                          uint16_t type_mask, HashEntry *hash_entry,
                          DataEntry *data_entry, HashEntry **entry_base,
-                         bool search_for_write) {
-  assert(!search_for_write || local_thread.id >= 0);
+                         SearchPurpose purpose) {
+  assert(purpose == SearchPurpose::READ || local_thread.id >= 0);
```

Comment:
Since local_thread is only for write thread, can we rename it to write_thread_local ?
then  will be clear of this assertion: it's for read or it is a valid write.

Reply:
ok, I'll just call it write thread

Reply:
In next pr